### PR TITLE
Planet 5322 add shfmt

### PR DESCRIPTION
**Pull request by r0bobo**
This adds shfmt to the shellscript linting and all scrips in this repo have been auto-formatted.

I have added a format and format-sh target to the Makefile. Do we want to keep this?

Other than that I also used shfmt -f . to find files for ShellCheck since it can find files both based on filename and shebang (which now also includes the .githooks/pre-commit file in this repo).

Fixes greenpeace/planet4#72

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +98,14 @@ clean:
 		@rm -f README.md $(SRC)/circleci-base/Dockerfile
 		@$(MAKE) -C test clean
 
+format: format-sh
+
+format-sh:
+ifndef SHFMT
+$(error "shfmt is not installed: https://github.com/mvdan/sh")
```
Comment (27Bslash6) [inline Makefile:105]: Might want to indent this so it doesn't complain if not found locally while executing other targets

Comment (r0bobo) [reply Makefile]: Fixed!

**Comment by r0bobo** [comment]
I've added the bats test and rebased on master to fix the conflicts.

**Comment by 27Bslash6** [comment]
Alright it took me a while (too long) to decipher why CirceCI checks aren't being applied... this PR is from a fork. @comzeradd @jen-cub can we get this contributor access to push to branches in relevant infrastructure repositories.  Recent changes to the workflows have actually made external contributions to this repository more brittle, as 

Another note, while there are branch protections in place for master, no conditions are set.  Should we at least enforce PR restrictions?

**Comment by comzeradd** [comment]
CI on forks was disabled on CircleCI. I just enabled it. Although it may still not work as expected, since I guess some CI jobs require secrets (which are not shared with forks). To avoid having ad-hoc permissions we can push the branch ourselves to the main repo when code review is ready, to do a final check by triggering tests to run. I just did for this branch so we'll see the CI results here on this PR.

**Comment by comzeradd** [comment]
> Another note, while there are branch protections in place for master, no conditions are set. Should we at least enforce PR restrictions?

Yes, we should. Right now the protection just prevents an accidental `push --force` to `master`. But we can also require CI pipelines to pass.

**Comment by 27Bslash6** [comment]
Now CI is failing due to the chicken/egg scenario.  One option to resolve is a build/push :latest tag locally, which is similar to how I've handled this in past, though again that was with :develop.  Another would be to change Makefile targets that rely on shfmt to use 
```Makefile
ifndef SHFMT
  $(warning ...)
else
  $(SHFMT) etc
fi
```

Other thoughts?

**Comment by r0bobo** [comment]
> Now CI is failing due to the chicken/egg scenario. One option to resolve is a build/push :latest tag locally, which is similar to how I've handled this in past, though again that was with :develop. Another would be to change Makefile targets that rely on shfmt to use
> 
> ```makefile
> ifndef SHFMT
>   $(warning ...)
> else
>   $(SHFMT) etc
> fi
> ```
> 
> Other thoughts?

Feels like a simple enough solution. I guess we could just enable the error again after it has successfully built if we want to?

**Pull request merged**